Pipe_Line_Execution.py

```diff
@@ -9,24 +9,31 @@ def run_clean(data_folder_path, processed_folder_path=False):
     imputed_frame = None
     file_index = 1
     for path in files_path:
-        frame = clu.load_and_first_digest_data(path)
+        frame = clu.load_and_first_digest_data(path) 
         if frame.empty:
             print(f"Empty DataFrame, check encoding or records existence in pre processed data")
             continue
 
         # Process
         initial_clean_frame = pro.clean_non_info_col(frame)
-        transformed_frame = pro.transform_frame(initial_clean_frame)
+        transformed_frame = pro.transform_frame(initial_clean_frame) # TODO: can you change the name of this function to something more meaningful?
         imputed_frame = pro.imputation(transformed_frame)
 
         # Optional save
         if processed_folder_path:
             clu.save_to_folder(imputed_frame, "clean_frame, " + f'{file_index}', processed_folder_path)
             file_index += 1
 
-    return imputed_frame
+    return imputed_frame # not sure why are you processing each file but return only the last
```

Comment:
don't sure I understand? Im processing in steps and for each I save the result, and process the result.
your meaning is that data frames are mutable so I can just run the pipe line and return frame var it self ?